src/main/java/hudson/plugins/ansicolor/AnsiColorBuildWrapper.java

```diff
@@ -117,19 +115,139 @@ private void addAll(AnsiColorMap[] maps, Map<String, AnsiColorMap> to) {
+                    validateFieldColorLiteral(colorMap.getCyan(), "cyan");
+                    validateFieldColorLiteral(colorMap.getCyanB(), "cyanB");
+                    validateFieldColorLiteral(colorMap.getWhite(), "white");
+                    validateFieldColorLiteral(colorMap.getWhiteB(), "whiteB");
+                }
+                final String globalColorMapName = req.getSubmittedForm().getString("globalColorMapName").trim();
+                final FormValidation validation = doCheckGlobalColorMapName(globalColorMapName);
+                if (validation.kind != FormValidation.Kind.OK) {
+                    throw new FormException(validation.getMessage(), "globalColorMapName");
+                }
+
+                if (!globalColorMapName.isEmpty() && colorMaps.stream().noneMatch(cm -> cm.getName().equals(globalColorMapName))) {
+                    throw new FormException("Global color map name must match one of the color maps", "globalColorMapName");
+                }
+                setColorMaps(colorMaps.toArray(new AnsiColorMap[0]));
+                setGlobalColorMapName(globalColorMapName.isEmpty() ? null : globalColorMapName);
                 save();
                 return true;
             } catch (ServletException e) {
                 throw new FormException(e, "");
             }
         }
 
+        private void validateFieldName(String fieldValue) throws FormException {
+            final FormValidation validation = doCheckName(fieldValue);
+            if (validation.kind != FormValidation.Kind.OK) {
+                throw new FormException(validation.getMessage(), "name");
+            }
+        }
+
+        private void validateFieldColorLiteral(String fieldValue, String fieldName) throws FormException {
+            final FormValidation globalColorMapNameValidation = validateColorLiteral(fieldValue);
+            if (globalColorMapNameValidation.kind != FormValidation.Kind.OK) {
+                throw new FormException(globalColorMapNameValidation.getMessage(), fieldName);
+            }
+        }
+
+        public FormValidation doCheckGlobalColorMapName(@QueryParameter String value) {
+            return value.isEmpty() ? FormValidation.ok() : validateColorMapName(value);
+        }
+
         @SuppressWarnings("unused")
         public FormValidation doCheckName(@QueryParameter final String value) {
-            return (value.trim().length() == 0) ? FormValidation.error("Name cannot be empty.") : FormValidation.ok();
+            return validateColorMapName(value);
+        }
+
+        private FormValidation validateColorMapName(String name) {
+            final int nameLength = name.trim().length();
+            return (nameLength < 1 || nameLength > 256) ? FormValidation.error("Color map name length must be between 1 and 256 chars.") : FormValidation.ok();
+        }
+
+        public FormValidation doCheckBlack(@QueryParameter String value) {
```

Comment:
How do these unused ones get invoked?

Reply:
That's Jenkins magic. All the `doCheck*` methods get called automatically on form data submission for every field with a matching method.